packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4455,7 +4455,8 @@ abstract class Element extends DiagnosticableTree implements BuildContext {
     if (_dependencies != null && _dependencies!.isNotEmpty) {
       final List<DiagnosticsNode> diagnosticsDependencies = _dependencies!
         .map((InheritedElement element) => element.widget.toDiagnosticsNode(style: DiagnosticsTreeStyle.sparse))
-        .toList();
+        .toList()
+        ..sort((DiagnosticsNode a, DiagnosticsNode b) => '$a'.compareTo('$b'));
```

Comment:
the toString for a DiagnosticsNode can be long and expensive (although shouldn't really be in this case). I would just use
toStringShort() instead to be safe toi make sure we don't end up doing O(n log n) computations of the full toString version of a diagnostic.